plugins/lookup/datacenter_moid.py

```diff
@@ -0,0 +1,66 @@
+# Copyright: (c) 2021, Alina Buzachis <@alinabuzachis>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+lookup: datacenter_moid
+short_description: Look up MoID for vSphere datacenter objects using vCenter REST API
+description:
+    - Returns Managed Object Reference (MoID) of the vSphere datacenter object contained in the specified path.
+author:
+    - Alina Buzachis <@alinabuzachis>
+version_added: 2.0.1
+requirements:
+    - vSphere 7.0.2 or greater
+    - python >= 3.6
+    - aiohttp
+extends_documentation_fragment:
+- vmware.vmware_rest.moid
+'''
+
+
+EXAMPLES = r'''
+# lookup sample
+- name: set connection info
+  set_fact:
+    connection_args:
+        vcenter_hostname: "vcenter.test"
+        vcenter_username: "administrator@vsphere.local"
+        vcenter_password: "1234"
+
+- name: lookup MoID of the object
+  debug: msg="{{ lookup('vmware.vmware_rest.datacenter_moid', '/my_dc', **connection_args) }}"
+
+- name: lookup MoID of the object inside the path
+  debug: msg="{{ lookup('vmware.vmware_rest.datacenter_moid', '/') }}"
```

Comment:
Here it the same thing. If you have multiple datacenters it errors you because you have to select one. When only one datacenter is available in `/` it returns the moid on that one.

Reply:
I changed it to make it even more clear.